contracts/staking/Staking.sol

```diff
@@ -372,59 +337,289 @@ contract Staking is
+                componentId = info.riskpoolId;
+            }
 
-        // TODO potentially reduce time depending on the time when the bundle has been closed
+            data = "";
+        } else {
+            revert("ERROR:STK-220:TARGET_TYPE_UNSUPPORTED");
+        }
 
-        rewardsAmount = calculateRewards(stakeInfo.balance, timeSinceLastUpdate);
+        // set target token where available
+        if(_registry.isRegisteredComponent(instanceId, componentId)) {
+            IComponentDataProvider.ComponentInfo memory info = _registry.getComponentInfo(instanceId, componentId);
+            token = info.token;
+            chainId = info.chainId;
+        }
+
+        return (
+            _toTargetId(targetType, instanceId, componentId, bundleId, data),
+            Target(
+                targetType,
+                instanceId,
+                componentId,
+                bundleId,
+                data,
+                token,
+                chainId
+            ));
     }
 
 
-    function calculateRewards(
-        uint256 amount,
-        uint256 duration
-    ) 
-        public override
+    function hasInfo(
+        bytes32 targetId,
+        address user
+    )
+        external override
         view
-        returns(uint256 rewardAmount) 
+        returns(bool hasStakeInfo)
     {
-        uint256 yearFraction = _math.itof(duration) / YEAR_DURATION;
-        uint256 rewardDuration = _math.mul(_rewardRate, yearFraction);
-        rewardAmount = _math.ftoi(amount * rewardDuration);
+        return _stakeInfo[targetId][user].createdAt > 0;
     }
 
-    function getStakingWallet()
-        public override
+
+    function getInfo(
+        bytes32 targetId,
+        address user
+    )
+        external override
         view
-        returns(address stakingWallet)
+        onlyInfo(targetId, user)
+        returns(StakeInfo memory info)
     {
-        return _stakingWallet;
+        return _stakeInfo[targetId][user];
     }
 
 
-    function oneYear() external override pure returns(uint256 yearInSeconds) {
-        return YEAR_DURATION;
+    function isStakingSupported(bytes32 targetId) 
+        external override
+        view 
+        onlyTarget(targetId)
+        returns(bool isSupported)
+    {
+        Target memory target = this.getTarget(targetId);
+
+        // currently only bundle staking is supported
+        if(target.targetType != TargetType.Bundle) {
+            return false;
+        }
+
+        IBundleDataProvider.BundleInfo memory info = _registry.getBundleInfo(target.instanceId, target.bundleId);
+        isSupported = false;
+
+        if(block.timestamp < info.expiryAt) {
+            if(info.closedAt == 0) {
+                isSupported = true;
+            } else if(block.timestamp < info.closedAt) {
+                isSupported = true;
+            }
+        }
+    }
+
+
+    function isUnstakingSupported(bytes32 targetId) 
```

Comment:
i think we should not here that unstaking is only supported when the target is closed/expired.

Reply:
for now we only support bundle targets. the idea was to return false for anything else than bundles. so far a concept is missing to define which target type has which conditions to allow staking/unstaking